src/main/java/teammates/logic/core/EmailGenerator.java

```diff
@@ -87,6 +89,87 @@
     }
     
     /**
+     * Generates the summary of the feedback sessions (in which at least one
+     * email has been sent to students) email for the given {@code courseId} for {@code student}
+     */
+    public EmailWrapper generateFeedbackSessionResendAllLinksEmail(String courseId,
+            StudentAttributes student) {
+        
+        CourseAttributes course = coursesLogic.getCourse(courseId);
+        
+        List<FeedbackSessionAttributes> sessions = new ArrayList<FeedbackSessionAttributes>();
+        List<FeedbackSessionAttributes> fsInCourse = fsDb.getFeedbackSessionsForCourse(courseId);
+        
+        for (FeedbackSessionAttributes fsa : fsInCourse) {
+            if (!fsa.isPrivateSession() && (fsa.isSentOpenEmail() || fsa.isSentPublishedEmail())) {
+                sessions.add(fsa);
+            }
+        }
+        
+        String submitUrl = null;
+        String reportUrl = null;
+        
+        StringBuffer linksFragmentValue = new StringBuffer(1000);
+        String joinUrl = Config.getAppUrl(student.getRegistrationUrl()).toAbsoluteString();
+        
+        String joinFragmentValue = "";
+        
+        if (isYetToJoinCourse(student)) {
+            joinFragmentValue = Templates.populateTemplate(EmailTemplates.FRAGMENT_STUDENT_COURSE_JOIN,
+                    "${joinUrl}", joinUrl,
+                    "${courseName}", course.getName());
+        }
+        
+        for (FeedbackSessionAttributes fsa : sessions) {
+            if (fsa.isOpened()) {
+                submitUrl = Config.getAppUrl(Const.ActionURIs.STUDENT_FEEDBACK_SUBMISSION_EDIT_PAGE)
+                        .withCourseId(course.getId())
+                        .withSessionName(fsa.getFeedbackSessionName())
+                        .withRegistrationKey(StringHelper.encrypt(student.key))
+                        .withStudentEmail(student.email)
+                        .toAbsoluteString();
+            } else {
+                submitUrl = "{Feedback session is closed}";
+            }
+
+            if (fsa.isPublished()) {
+                reportUrl = Config.getAppUrl(Const.ActionURIs.STUDENT_FEEDBACK_RESULTS_PAGE)
+                        .withCourseId(course.getId())
+                        .withSessionName(fsa.getFeedbackSessionName())
+                        .withRegistrationKey(StringHelper.encrypt(student.key))
+                        .withStudentEmail(student.email)
+                        .toAbsoluteString();
+            } else {
+                reportUrl = "{Feedback session is unpublished}";
```

Comment:
Similarly here:
```java
String reportUrl = fsa.isPublished()
                   ? Config.getAppUrl(...)
                   : "{Feedback session is not yet published}";
```
In addition, do not use "unpublish" because we use it to refer to an action.